### PR TITLE
fix: Observations requests interfaces

### PR DESCRIPTION
**Pull request by raop155**

### CODE REVIEW

File: src/interfaces.ts [hunk 1]
```diff
@@ -260,14 +260,35 @@ export interface IObservationsInput extends TSortOptions {}
...
-  sequence?: number;
-  specific?: {
-    signer: string;
-    hash: string;
-  };
-}
+type TObservationsInputs =
+  | {
+      chainId: number;
+      emitter?: undefined;
```
Comment (sebastianscatularo) [inline src/interfaces.ts:266]: why just you do not omit the undefined attributes, the join type do the work if it is necessary

Comment (valentinoConti) [reply src/interfaces.ts:266]: Sometimes you need to define the undefined attributes for Typescript to understand it correctly, on this video example, I need a method to always receive `chainId` and from there I can receive `emitter` and from there (only if we have emitter) we can receive the `specific` parameter


https://user-images.githubusercontent.com/41705567/216125664-f4349d0f-76df-42aa-b3e2-4b1e6be9e00e.mov

(didn't analyze this PR tho, maybe we can reduce some blocks here

File: src/interfaces.ts [hunk 2]
```diff
@@ -260,14 +260,35 @@ export interface IObservationsInput extends TSortOptions {}
...
+      sequence?: undefined;
+      specific?: undefined;
+    }
+  | {
+      chainId: number;
+      emitter: string;
+      sequence: number;
+      specific?: undefined;
+    }
+  | {
```
Comment (valentinoConti) [inline src/interfaces.ts:282]: this last block here can be removed if you replace on the block above
```ts
specific?: undefined;
```
for
```ts
specific?: {
  signer: string;
  hash: string;
}
```

Comment (raop155) [reply src/interfaces.ts]: done! 👍

**Review by valentinoConti — APPROVED** [review]

**Pull request merged**